### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by bsr203**
hi,

The first commit is working fine. I then tried to use `redux-act` but there is an issue on how I register the action with store as the reducers are not triggered on action. Can you please check. Also, you may comment on placement of new files.

Cheers,
bsr

### CODE REVIEW

**Comment by bsr203** [comment]
just noticed the test is failing. I only use unit tests, and not sure why they fails. Probably because I added new dom elements. can you please comment.

**Comment by bsr203** [comment]
to test, you can stop neo4j and which shows an `important` notification, which needs explicit acknowledgement. There is also a not important welcome message when we load the page.

**Comment by bsr203** [comment]
I didn't spend  much time on it to figure out the failing test and `redux-act` thinking your expertize can save some. What do you think about the it otherwise.

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "passport-http-bearer": "^1.0.1",
     "react": "^0.14.3",
     "react-dom": "^0.14.3",
+    "react-icons": "^1.0.3",
```
Comment (vslinko) [inline package.json:62]: I'm not sure that boilerplate should depend on some style-specific library.

File: src/frontend/bundles/notification/components/Notification/Notification.less [hunk 2]
```diff
@@ -0,0 +1,108 @@
+ /*
+ adapted from https://github.com/7rulnik/evil-martians-test-project
+ no license specified
+ */
+
+.notification {
```
Comment (vslinko) [inline src/frontend/bundles/notification/components/Notification/Notification.less:6]: Also I don't want to add predefined styles into boilerplate.

File: src/frontend/bundles/notification/components/Notification/Notification.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        { viewUnread &&
+          <div className={styles.date} >{day}.{month}.{year}</div>
+        }
+      </div>
+    );
+  }
+}
+
+Notification.propTypes = {
```
Comment (vslinko) [inline src/frontend/bundles/notification/components/Notification/Notification.js:63]: `propTypes` and `defaultProps` should be defined inside the class using `static` syntax.

File: src/frontend/bundles/notification/reducers/notifications.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    [ac.readAllNotification]: (state, {}) => {
+      return state.map(notification => notification.important ?
+        Object.assign({}, notification, { read: true }) : notification
+      );
+    },
+  },
+  [{
+    text: 'Welcome to ripster!',
+    important: false,
```
Comment (vslinko) [inline src/frontend/bundles/notification/reducers/notifications.js:34]: Important isn't important for the boilerplate :smile:
Boilerplate should be as simple as possible.

File: src/frontend/bundles/notification/reducers/notifications.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    id: 1,
+    read: false,
```
Comment (vslinko) [inline src/frontend/bundles/notification/reducers/notifications.js:36]: Viewed notifications should be deleted from the store.
Could you remove this property and change the reducer?

File: src/frontend/initApp.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { createHistory } from 'history';
+import createHistory from 'history/lib/createBrowserHistory';
```
Comment (vslinko) [inline src/frontend/initApp.js:1]: Could you create a separate PR with those changes?

File: src/frontend/initApp.js [hunk 7]
```diff
@@ -9,19 +9,28 @@ import { RelayRoutingContext } from 'react-router-relay';
...
+    store.dispatch(addNotification({
+      important: true,
+      text: error.message,
+    }));
+  };
+
   try {
     Relay.injectNetworkLayer(
-      new Relay.DefaultNetworkLayer('/graphql', {
+      new RelayNetworkLayer('/graphql', {
```
Comment (vslinko) [inline src/frontend/initApp.js:29]: I think here should be more flexible API. For example we could merge fetch options with other network layer options:

``` js
new CustomNetworkLayer('/graphql', {
  fetchOptions: {
    credentials: 'same-origin',
  },
  onError,
});
```

File: src/frontend/RelayNetworkLayer.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    return this._sendMutation(request).then(
+      result => result.json()
+    ).then(payload => {
+      if (payload.hasOwnProperty('errors')) {
+        const error = new Error(
+          'Server request for mutation `' + request.getDebugName() + '` ' +
+          'failed for the following reasons:\n\n' +
+          this.formatErrors(request, payload.errors)
+        );
+        (error: any).source = payload;
```
Comment (vslinko) [inline src/frontend/RelayNetworkLayer.js:49]: We don't use `flow` yet. So, could you remove all the flow annotations?

**Comment by vslinko** [comment]
Thank you for your PR. First batch of comments is ready.

**Comment by bsr203** [comment]
Hi @vslinko. thanks for your effort on detailed review. I was bit busy last few days. I will finish it when I started using this part again, and hope it helps someone as an example till then. thanks again.

**Pull request closed without merging**